Tools/clinic/clinic.py

```diff
@@ -2917,16 +2921,15 @@ def render(self, parameter: Parameter, data: CRenderData) -> None:
         self._render_self(parameter, data)
         self._render_non_self(parameter, data)
 
-    def length_name(self):
+    def length_name(self) -> str:
         """Computes the name of the associated "length" variable."""
-        if not self.length:
-            return None
+        assert self.length is not None
         return self.parser_name + "_length"
 
     # Why is this one broken out separately?
     # For "positional-only" function parsing,
     # which generates a bunch of PyArg_ParseTuple calls.
-    def parse_argument(self, list):
+    def parse_argument(self, list: list[str]) -> None:
```

Comment:
Okay but let's change the parameter name while we're here -- `list: list` looks ridiculous :P

Reply:
I did that in an earlier version! 😆

Reply:
How about simply `args`?

Reply:
Works for me!